internal/provider/vdc_resource.go

```diff
@@ -0,0 +1,480 @@
+	ID                     types.String             `tfsdk:"id"`
+	Name                   types.String             `tfsdk:"name"`
+	Description            types.String             `tfsdk:"description"`
+	VdcServiceClass        types.String             `tfsdk:"vdc_service_class"`
+	VdcDisponibilityClass  types.String             `tfsdk:"vdc_disponibility_class"`
+	VdcBillingModel        types.String             `tfsdk:"vdc_billing_model"`
+	VcpuInMhz2             types.Float64            `tfsdk:"vcpu_in_mhz2"`
+	CPUAllocated           types.Float64            `tfsdk:"cpu_allocated"`
+	MemoryAllocated        types.Float64            `tfsdk:"memory_allocated"`
+	VdcStorageBillingModel types.String             `tfsdk:"vdc_storage_billing_model"`
+	VdcStorageProfiles     []vdcStorageProfileModel `tfsdk:"vdc_storage_profiles"`
+	VdcGroup               types.String             `tfsdk:"vdc_group"`
+}
+
+type vdcStorageProfileModel struct {
+	Class   types.String `tfsdk:"class"`
+	Limit   types.Int64  `tfsdk:"limit"`
+	Default types.Bool   `tfsdk:"default"`
+}
+
+// Metadata returns the resource type name.
+func (r *vdcResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vdc"
+}
+
+// Schema defines the schema for the resource.
+func (r *vdcResource) Schema(ctx context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "VDC resource allows you to create a org VDC in Cloud Avenue.",
+		Attributes: map[string]schema.Attribute{
+			"timeouts": timeouts.Attributes(ctx, timeouts.Opts{
+				Create: true,
+				Read:   true,
+			}),
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The name of the org VDC. It must be unique in the organization. The length must be between 2 and 27 characters.",
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(2, 27),
+				},
+			},
+			"description": schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "The description of the org VDC.",
+			},
+			"vdc_group": schema.StringAttribute{
+				Required: true,
+				MarkdownDescription: "Name of an existing vDC group or a new one. This allows you to isolate your vDC." +
+					"VMs of vDCs which belong to the same vDC group can communicate together.",
+			},
+			"vcpu_in_mhz2": schema.Float64Attribute{
+				Required: true,
+				MarkdownDescription: "Specifies the clock frequency, in Mhz, for any virtual CPU that is allocated to a VM." +
+					"It must be at least 1200.",
+				Validators: []validator.Float64{
+					float64validator.AtLeast(1200),
+				},
+			},
+			"cpu_allocated": schema.Float64Attribute{
+				Required: true,
+				MarkdownDescription: "Capacity that is committed to be available or used as a limit in PAYG mode." +
```

Comment:
Please add `\n` at end line otherwise in the generated markdown this is rended like that

```markdown
(...) as a limit in PAYG mode.Unit for compute (...)
```